model/distro/distro.go

```diff
@@ -194,7 +204,23 @@ func ValidateContainerPoolDistros(s *evergreen.Settings) error {
 	return errors.WithStack(catcher.Resolve())
 }
 
-// ValidateBootstrapMethod ensure that the bootstrap mechanism is one of the
+// ValidateArch checks that the architecture is one of the supported
+// architectures.
+func ValidateArch(arch string) error {
+	osAndArch := strings.Split(arch, "_")
+	if len(osAndArch) != 2 {
+		return fmt.Errorf("architecture '%s' is not in the form ${GOOS}_${GOARCH}", arch)
+	}
+
+	switch arch {
+	case ArchDarwinAmd64, ArchLinux386, ArchLinuxPpc64le, ArchLinuxS390x, ArchLinuxArm64, ArchLinuxAmd64, ArchWindows386, ArchWindowsAmd64:
```

Comment:
It might be cleaner to have a slice of valid architectures and check if the slice contains the one you're checking